docs/map.xsd

```diff
@@ -204,41 +358,92 @@
-  <xs:attribute name="tilewidth" type="xs:nonNegativeInteger" use="required"/>
-  <xs:attribute name="tileheight" type="xs:nonNegativeInteger" use="required"/>
+  <xs:attribute name="tilewidth" type="pixelT" use="required"/>
+  <xs:attribute name="tileheight" type="pixelT" use="required"/>
+  <xs:attribute name="hexsidelength" type="pixelT" use="optional"/>
+  <xs:attribute name="staggeraxis" type="staggeraxisT" use="optional"/>
+  <xs:attribute name="staggerindex" type="staggerindexT" use="optional"/>
+  <xs:attribute name="backgroundcolor" type="hexColourAlphaT" use="optional"/>
+  <xs:attribute name="nextobjectid" type="xs:nonNegativeInteger" use="required"/>
 </xs:attributeGroup>
 
 <xs:attributeGroup name="tileset">
   <xs:attribute name="name" type="nameT"/>
   <xs:attribute name="firstgid" type="xs:positiveInteger" use="required"/>
   <xs:attribute name="source" type="xs:anyURI"/>
-  <xs:attribute name="tilewidth" type="xs:nonNegativeInteger"/>
-  <xs:attribute name="tileheight" type="xs:nonNegativeInteger"/>
-  <xs:attribute name="spacing" type="xs:nonNegativeInteger"/>
+  <xs:attribute name="tilewidth" type="pixelT"/>
+  <xs:attribute name="tileheight" type="pixelT"/>
+  <xs:attribute name="spacing" type="pixelT"/>
+  <xs:attribute name="margin" type="pixelT"/>
+  <xs:attribute name="tilecount" type="xs:nonNegativeInteger"/>
+  <xs:attribute name="columns" type="xs:nonNegativeInteger"/>
 </xs:attributeGroup>
 
-<xs:attributeGroup name="image.tileset">
+  <xs:attributeGroup name="imagelayer">
+    <xs:attribute name="name" type="nameT"/>
+    <xs:attribute name="offsetx" type="xs:integer" default="0" use="optional"/>
+    <xs:attribute name="offsety" type="xs:integer" default="0" use="optional"/>
+    <xs:attribute name="opacity" type="opacityT" default="1" use="optional"/>
+    <xs:attribute name="visible" type="xs:boolean" default="1" use="optional"/>
+  </xs:attributeGroup>
+
+  <xs:attributeGroup name="terrain">
+    <xs:attribute name="name" type="nameT"/>
+    <xs:attribute name="tile" type="tileId"/>
+  </xs:attributeGroup>
+
+  <xs:attributeGroup name="tileoffset">
+    <xs:attribute name="x" type="xs:integer" use="required"/>
+    <xs:attribute name="y" type="xs:integer" use="required"/>
+  </xs:attributeGroup>
+
+  <xs:attributeGroup name="data">
+    <xs:attribute name="encoding" type="encodingT" use="required"/>
+    <xs:attribute name="compression" type="compressionT" use="required"/>
+  </xs:attributeGroup>
+
+  <xs:attributeGroup name="frame">
+    <xs:attribute name="tileid" type="tileId" use="required"/>
+    <xs:attribute name="duration" type="xs:nonNegativeInteger" use="required"/>
+  </xs:attributeGroup>
+  
+  <xs:attributeGroup name="image">
+    <xs:attribute name="format" type="imageformatT" use="required"/>
+    <xs:attribute name="source" type="xs:anyURI" use="required"/>
+    <xs:attribute name="trans" type="hexColourT" use="required"/>
+    <xs:attribute name="width" type="pixelT" use="required"/>
```

Comment:
None of these attributes are required. Though, you would need to have at least `format` (for embedded data) or `source` (for external data), otherwise there is no image. Anyway I would mark both as optional if there is no easy way to represent that.